WordPress/WordPressShare/ShareViewController.swift

```diff
@@ -0,0 +1,156 @@
+            let configuration = NSURLSessionConfiguration.backgroundSessionConfigurationWithIdentifier(WPAppGroupName)
+            configuration.sharedContainerIdentifier = WPAppGroupName
+            let service = PostService(configuration: configuration)
+            let subjectAndBody = self.splitContentTextIntoSubjectAndBody(self.contentWithSourceURL(url))
+            service.createPost(siteID: self.selectedSiteID!, status:self.postStatus, title: subjectAndBody.subject, body: subjectAndBody.body) { (post, error) -> Void in
+                print("Post \(post) Error \(error)")
+            }
+            
+            self.extensionContext!.completeRequestReturningItems([], completionHandler: nil)
+        }
+    }
+
+    override func configurationItems() -> [AnyObject]! {
+        let blogPickerItem = SLComposeSheetConfigurationItem()
+        blogPickerItem.title = NSLocalizedString("Post to:", comment: "Upload post to the selected Site")
+        blogPickerItem.value = selectedSiteName ?? NSLocalizedString("Select a site", comment: "Select a site in the share extension")
+        blogPickerItem.tapHandler = { [weak self] in
+            self?.displayBlogPicker()
+        }
+        
+        let statusPickerItem = SLComposeSheetConfigurationItem()
+        statusPickerItem.title = NSLocalizedString("Post Status:", comment: "Post status picker title in Share Extension")
+        statusPickerItem.value = self.postStatuses[postStatus]!
+        statusPickerItem.tapHandler = { [weak self] in
+            self?.displayStatusPicker()
+        }
+        
+        return [blogPickerItem, statusPickerItem]
+    }
+
+    
+    private func displayBlogPicker() {
+        let pickerViewController = BlogPickerViewController()
+        pickerViewController.onChange = { (siteId, description) in
+            self.selectedSiteID = siteId
+            self.selectedSiteName = description
+            self.reloadConfigurationItems()
+        }
+        
+        pushConfigurationViewController(pickerViewController)
+    }
+    
+    private func displayStatusPicker() {
+        let pickerViewController = PostStatusPickerViewController()
+        pickerViewController.statuses = postStatuses
+        pickerViewController.onChange = { (status, description) in
+            self.postStatus = status
+            self.reloadConfigurationItems()
+        }
+        
+        pushConfigurationViewController(pickerViewController)
+    }
+    
+    
+    // TODO: This should eventually be moved into WordPressComKit
+    private let postStatuses = [
+        "draft" : NSLocalizedString("Draft", comment: "Draft post status"),
+        "publish" : NSLocalizedString("Publish", comment: "Publish post status")]
+
+    private func splitContentTextIntoSubjectAndBody(contentText: String) -> (subject: String, body: String) {
+        let fullText = contentText
+        let firstCarriageReturnIndex = fullText.rangeOfCharacterFromSet(NSCharacterSet.newlineCharacterSet())
```

Comment:
As nitpick as it gets, but this isn't Windows :stuck_out_tongue:
How about `indexOfFirstNewline`?